### DISCUSSION TITLE
Consider renaming x-data

### DISCUSSION THREAD

**Discussion by james0r** [alpinejs/alpine#1886] [3. Feature ideas]
I just want to propose renaming this directive to something like `x-component` for the following reasons.

1) Unlike other directives, its name refers to what values it accepts rather than the directive's primary function.
2) With the addition of the Alpine store in v3, `x-data` is required to initialize an element as an Alpine component in order to use state, but doesn't require any state to be passed as a value to the new component.

Seems like I had a 3rd reason before but I can't remember at the moment, but the the 3rd reason notwithstanding, I think these two reasons are pretty compelling.

Thanks for the consideration! Really enjoying Alpine!

**Comment by KevinBatdorf** [comment]
I think `x-data` is pretty much a big part of Alpine's branding at this point, although as a user, I personally agree with your points. Curious what Caleb and others think though.

**Comment by luizbills** [comment]
Maybe `x-app` with `x-data` as alias.
